packages/apollo-link-dedup/src/dedupLink.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  ApolloLink,
+  Operation,
+  NextLink,
+  FetchResult,
+  Observable,
+} from 'apollo-link-core';
+
+import { print } from 'graphql/language/printer';
+
+/*
+ * Expects context to contain the deduplicate field
+ */
+export class DedupLink extends ApolloLink {
+  private inFlightRequestPromises: { [key: string]: Observable<FetchResult> };
+
+  constructor() {
+    super();
+    this.inFlightRequestPromises = {};
+  }
+
+  public request(
+    operation: Operation,
+    forward: NextLink,
+  ): Observable<FetchResult> {
+    // sometimes we might not want to deduplicate a request, for example when we want to force fetch it.
```

Comment:
I think it's better to opt-out since the user is including it in their composition, so I switched it up to `forceFetch`. Hoping someone can come up with a better name 🤞